site/src/modules/resources/AgentLogs/useAgentLogs.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { act, renderHook, waitFor } from "@testing-library/react";
+import WS from "jest-websocket-mock";
+import { type QueryClient, QueryClientProvider } from "react-query";
+import { API } from "api/api";
+import * as APIModule from "api/api";
+import { agentLogsKey } from "api/queries/workspaces";
+import type {
+  WorkspaceAgentLifecycle,
+  WorkspaceAgentLog,
+} from "api/typesGenerated";
+import { MockWorkspace, MockWorkspaceAgent } from "testHelpers/entities";
+import { createTestQueryClient } from "testHelpers/renderHelpers";
+import { type UseAgentLogsOptions, useAgentLogs } from "./useAgentLogs";
+
+afterEach(() => {
+  WS.clean();
+});
+
+describe("useAgentLogs", () => {
+  it("should not fetch logs if disabled", async () => {
+    const queryClient = createTestQueryClient();
+    const fetchSpy = jest.spyOn(API, "getWorkspaceAgentLogs");
+    const wsSpy = jest.spyOn(APIModule, "watchWorkspaceAgentLogs");
+    renderUseAgentLogs(queryClient, "ready", { enabled: false });
+    expect(fetchSpy).not.toHaveBeenCalled();
+    expect(wsSpy).not.toHaveBeenCalled();
```

Comment:
I forget how the render helpers work: just to make sure, could you try enabling this, and see if the tests still pass?

My main worry is that I can't remember if React Query will actually kick off a request right away, or if it needs a little bit of time. And if it's a liltle bit of time, could the tests finish up before they have a chance to fail?

Reply:
I have tests to check the WebSocket connection right after this one.